### PR TITLE
[1.2.x] hardening: remove PHP_EOL from force_https redirect header

### PR DESCRIPTION
**Pull request by somethingwithproof**
## Summary
- Remove `PHP_EOL` from Location header value
- Tighten HTTPS detection to handle edge cases (`$_SERVER['HTTPS']` set to empty or 'off')

Backport of #6659.

## Test plan
- [ ] Verify force_https redirect works correctly

### CODE REVIEW

File: include/global.php [hunk 1]
```diff
@@ -415,8 +415,10 @@
 
 if ($config['is_web']) {
 	if (read_config_option('force_https') == 'on') {
-		if (!isset($_SERVER['HTTPS']) && isset($_SERVER['HTTP_HOST']) && isset($_SERVER['REQUEST_URI'])) {
-			header('Location: https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'] . PHP_EOL . PHP_EOL);
+		$is_https = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== '' && strtolower($_SERVER['HTTPS']) !== 'off');
+
+		if (!$is_https && isset($_SERVER['HTTP_HOST']) && isset($_SERVER['REQUEST_URI'])) {
+			header('Location: https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI']);
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply include/global.php:421]: Valid observation. Unifying the HTTPS detection is out of scope for this backport; filed as a follow-up.

**Review by TheWitness — APPROVED** [review]
